common.sh

```diff
@@ -32,6 +47,24 @@ get_svn_revision() {
   [ ! -e $TO_DIR ] && svn co -r$SVN_REVISION $SVN_REPO $TO_DIR
 }
 
+build_afl() {
+  $CC -c -w $AFL_SRC/llvm_mode/afl-llvm-rt.o.c
+  $CXX -g -fsanitize-coverage=trace-pc-guard $TARGET_FILE $TARGET_INCLUDE -c
+
+  UNIQUE_BUILD="$AFL_DRIVER afl-llvm-rt.o.o ${TARGET_NAME}.o $UNIQUE_BUILD"
```

Comment:
again, you are using TARGET_NAME defined in build.sh. Please don't. 
More generally, avoid circular deps between .sh files -- it gets too messy too soon. 
both build_afl and build_libfuzzer should produce libFuzzingEngine.a or some such, then build.sh should link against it.